### ISSUE TITLE
Can't build indy-sdk on mac os get Rust Errors

### ISSUE THREAD

**Issue by droconnel22** [hyperledger/indy-sdk#2203]
Run the command:
`
./mac.build.sh
`
Mac Os Catalina version 10.15.5 (19F101)
xcode-select version 2373.
Homebrew 2.2.11
Homebrew/homebrew-core (git revision 65af; last commit 2020-04-04)
Homebrew/homebrew-cask (git revision 647a2; last commit 2020-04-05)
cmake version 3.14.3
rustc 1.37.0 (eae3437df 2019-08-13)

```
To configure your current shell run source $HOME/.cargo/env
   Compiling indy-utils v0.1.0 (my-path)
error[E0008]: cannot bind by-move into a pattern guard
   --> indy-utils/src/wql.rs:114:23
    |
114 |             Query::In(key, ref targets) if targets.len() == 1 => {
    |                       ^^^ moves value into pattern guard

error[E0507]: cannot move out of `*operator` which is behind a shared reference
  --> indy-utils/src/wql.rs:86:46
   |
86 |                         .flat_map(|operator| operator.optimise())
   |                                              ^^^^^^^^ move occurs because `*operator` has type `wql::Query`, which does not implement the `Copy` trait

error[E0507]: cannot move out of `*operator` which is behind a shared reference
   --> indy-utils/src/wql.rs:105:46
    |
105 |                         .flat_map(|operator| operator.optimise())
    |                                              ^^^^^^^^ move occurs because `*operator` has type `wql::Query`, which does not implement the `Copy` trait

error[E0596]: cannot borrow `*suboperators` as mutable, as it is behind a `&` reference
  --> indy-utils/src/wql.rs:80:17
   |
79 |             Query::And(ref suboperators) if suboperators.len() == 1 => {
   |                        ---------------- help: consider changing this to be a mutable reference: `ref mut suboperators`
80 |                 suboperators.remove(0).optimise()
   |                 ^^^^^^^^^^^^ `suboperators` is a `&` reference, so the data it refers to cannot be borrowed as mutable

error[E0596]: cannot borrow `*suboperators` as mutable, as it is behind a `&` reference
  --> indy-utils/src/wql.rs:99:17
   |
98 |             Query::Or(ref  suboperators) if suboperators.len() == 1 => {
   |                       ----------------- help: consider changing this to be a mutable reference: `ref mut  suboperators`
99 |                 suboperators.remove(0).optimise()
   |                 ^^^^^^^^^^^^ `suboperators` is a `&` reference, so the data it refers to cannot be borrowed as mutable

error[E0596]: cannot borrow `*targets` as mutable, as it is behind a `&` reference
   --> indy-utils/src/wql.rs:115:37
    |
114 |             Query::In(key, ref targets) if targets.len() == 1 => {
    |                            ----------- help: consider changing this to be a mutable reference: `ref mut targets`
115 |                 Some(Query::Eq(key, targets.remove(0)))
    |                                     ^^^^^^^ `targets` is a `&` reference, so the data it refers to cannot be borrowed as mutable

error: aborting due to 6 previous errors

Some errors have detailed explanations: E0008, E0507, E0596.
For more information about an error, try `rustc --explain E0008`.
error: Could not compile `indy-utils`.

```

**Comment by droconnel22** [comment]
Did a checkout with v1.14.0 and I was able to build on Mac Os.

**Comment by msembinelli** [comment]
What version of rust are you using? Honestly I had massive issues until I updated with rustup to the nightly release.

**Comment by droconnel22** [comment]
@msembinelli  
I was using rustc 1.37.0 (eae3437df 2019-08-13) switched to rustc 1.46.0-nightly (f455e46ea 2020-06-20) and it worked. Thank you!

**Comment by msembinelli** [comment]
no problem @droconnel22 ! This should really be documented for each release

**Comment by dhh1128** [comment]
I'm reopening this to get a clarification. I am under the impression that it is an error to require Rust nightly to build -- that the SDK is supposed to build with standard Rust instead. I have been building the SDK regularly without having Rust nightly; it this has changed, I want to know.

**Comment by dhh1128** [comment]
Note the conversation about build problems in #2207

**Comment by Patrik-Stas** [comment]
@dhh1128 The pipeline issue was solved by updating to `1.40`, I am quite sure you don't need nightly. Shall we close this one?

**Comment by dhh1128** [comment]
Okay, great!
